src/pages/programDisaggregations/form/DissaggregationCategories.tsx

```diff
@@ -0,0 +1,460 @@
+
+const useProgramIndicatorSelectionCategories = (
+    categoriesWithMappings: string[]
+) => {
+    const programIndicatorsValue: ProgramIndicatorValue = useField(
+        'programIndicatorMappings'
+    )?.input?.value
+    const programCategories = Object.values(programIndicatorsValue)
+        ?.map((piValue) => {
+            return piValue?.categoryCombo?.categories
+        })
+        ?.flat()
+        ?.map((category) => category?.id)
+    return [...new Set(programCategories)].filter(
+        (id) => !categoriesWithMappings.includes(id)
+    )
+}
+
+export const DisaggregationCategories = () => {
+    const formApi = useForm()
+
+    // this is to have a stable-ish reference instead of taking from formState (rethink)
+    const { initialValues } = useFormState()
+    const [categoriesWithMappings, setCategoriesWithMappings] = useState<
+        string[]
+    >([])
+
+    useEffect(() => {
+        setCategoriesWithMappings(
+            Object.keys(initialValues?.categoryMappings || {})
+        )
+    }, [initialValues.categoryMappings])
+
+    const [addedCategories, setAddedCategories] = useState<string[]>([])
+
+    const categoriesFromProgramIndicatorSelections =
+        useProgramIndicatorSelectionCategories(categoriesWithMappings)
+
+    const queryFn = useBoundResourceQueryFn()
+    const query = {
+        resource: 'categoryCombos',
+        params: {
+            paging: false,
+            fields: [
+                'id',
+                'displayName',
+                'categories[id,displayName,categoryOptions[id,displayName]]',
+            ],
+            filters: ['name:neq:default'],
+        },
+    }
+    const categoryCombos = useQuery({
+        queryKey: [query],
+        queryFn: queryFn<CategoryComboResponse>,
+    })
+
+    const {
+        categoryCombosToCategoriesObject,
+        categoryObject,
+        categoriesArray,
+        categoryCombosArray,
+    } = useMemo(() => {
+        return getCategoryObjects(categoryCombos.data)
+    }, [categoryCombos.data])
+
+    const { show: showNotification } = useAlert((message) => message)
+
+    if (categoryCombos.isLoading) {
+        return null
+    }
+
+    if (categoryCombos?.data?.categoryCombos?.length === 0) {
+        return (
+            <div>
+                {i18n.t(
+                    'There are no category combinations. Your system must have at least one category combination in order to disaggregate a program indicator.'
+                )}
+            </div>
+        )
+    }
+
+    const addSingleCategory = (categoryId: string) => {
+        if (categoriesWithMappings.includes(categoryId)) {
+            return
+        }
+        const options = getEmptyOptionValues(
+            categoryObject?.[categoryId]?.categoryOptions
+        )
+
+        formApi.change(`categoryMappings.${categoryId}`, [
+            {
+                categoryId: categoryId,
+                id: generateDhis2Id(),
+                mappingName: 'Standard mapping',
```

Comment:
this is not translated because the metadata here is not translatable, so it seems reasonable to default it to English, regardless of user language. I suppose most correct would be to translate based on system's default language?